frontend/javascripts/oxalis/model/actions/skeletontracing_actions.tsx

```diff
@@ -596,20 +598,21 @@ function confirmDeletingInitialNode(treeId: number) {
 export const deleteTreeAsUserAction = (treeId?: number): NoAction => {
   const state = Store.getState();
   const skeletonTracing = enforceSkeletonTracing(state.annotation);
-  getTree(skeletonTracing, treeId).map((tree) => {
-    if (state.task != null && tree.nodes.has(1)) {
-      confirmDeletingInitialNode(tree.treeId);
-    } else if (state.userConfiguration.hideTreeRemovalWarning) {
-      Store.dispatch(deleteTreeAction(tree.treeId));
-    } else {
-      renderIndependently((destroy) => (
-        <RemoveTreeModal
-          onOk={() => Store.dispatch(deleteTreeAction(tree.treeId))}
-          destroy={destroy}
-        />
-      ));
-    }
-  });
+  const tree = getTree(skeletonTracing, treeId);
+  if (!tree) return noAction();
```

Comment:
Is `noAction()` the correct early return value here?

Reply:
this function is a bit weird, because it only returns `noAction` and then potentially dispatches other actions (async) on its own. however, the interface looks like it was a normal action creator. this was already weird before this pr.
my suggestion would be to rename the function to: `handleDeleteTreeByUser` and make it return nothing. you could do the same for `deleteNodeAsUserAction`.